File: schemaorg/warrantyscope.go
package schemaorg

import "encoding/json"

// WarrantyScope see : https://schema.org/WarrantyScope
type WarrantyScope struct {

typeContext

Enumeration

// SupersededBy see : https://schema.orghttp://meta.schema.org/supersededBy
// Relates a term (i.e. a property, class or enumeration) to one that supersedes it.
SupersededBy interface{} `json:"supersededBy"` // types : Class Enumeration Property

}

func (v *WarrantyScope) MarshalJSON() ([]byte, error) {
	v.C = "http://schema.org"
	v.T = "WarrantyScope"

	return json.Marshal(v)
}

File: schemaorgphp/classes/warranty_scope.php
<?php

class WarrantyScope extends Enumeration implements JsonSerializable {

	public static $context = 'http://schema.org';
	public static $type = 'WarrantyScope';
	
	/**
	 * Relates a term (i.e. a property, class or enumeration) to one that supersedes it.
	 * see : https://meta.schema.org/supersededBy
	 * @var \Class|\Class[]|\Enumeration|\Enumeration[]|\Property|\Property[]
	 */
	public var $superseded_by;
	
	public function jsonSerialize() {
		$out = array(
			'@context' => 'http://schema.org',
			'@type' => 'WarrantyScope'
		);
		
		$serialized = so_json_serialize( $this->superseded_by );
		if ( ! empty( $serialized ) ) {
			$out['supersededBy'] = $serialized;
		}
		
		return $out;
	}
}
